### PR TITLE
Philip - Homework3

### PR DESCRIPTION
**Pull request by EvanSimpson**
Graded by @thomasnat1 @EvanSimpson

### CODE REVIEW

File: classes/class3/homework/routes/cats.js [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+}
+
+function create(req, res) {
+  var color = colors[Math.floor(Math.random() * colors.length)];
+  var name = names[Math.floor(Math.random() * names.length)];
+  var age = Math.floor(Math.random() * name.length);
+  var catObject = {
+    name: name,
+    age: age,
+    colors: color
```
Comment (EvanSimpson) [inline classes/class3/homework/routes/cats.js:48]: `colors` is expecting 1) an array and 2) to be called `color`, see

``` js
var catSchema = mongoose.Schema({
  name: String,
  age: Number,
  color: [String]
});
```

in `cat.js`.

What you're doing will, in MongoDB, attempt to create a new attribute in the cat called `colors`, but Mongoose is preventing that from happening with its predefined schema.

File: classes/class3/homework/routes/cats.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+  var colorFilter;
+  Cat.find({colors: colorFilter}, function(err, cats) {
+    res.render('cats', {cats: cats});
+  });
+}
+
+function create(req, res) {
+  var color = colors[Math.floor(Math.random() * colors.length)];
+  var name = names[Math.floor(Math.random() * names.length)];
+  var age = Math.floor(Math.random() * name.length);
```
Comment (EvanSimpson) [inline classes/class3/homework/routes/cats.js:44]: Interesting choice, choosing the age based on name length? Also, this is somehow producing ages of only 0 or 1.

File: classes/class3/homework/routes/cats.js [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+}
+
+function list(req, res) {
+  var colorFilter;
+  if (req.params.color) {
+    colorFilter = req.params.color.toLowerCase();
+  } else {
+    colorFilter = '';
+  }
+  Cat.find({colors: colorFilter}).sort({age: -1}).exec(function(err, cats) {
```
Comment (EvanSimpson) [inline classes/class3/homework/routes/cats.js:67]: Despite the `.sort()` your cats are managing to not be sorted by age.

[Example](https://www.dropbox.com/s/symfwmlatlz9ihs/Screenshot%202015-02-03%2020.13.01.png?dl=0)

You've got some interesting things going on in your app here.

**Comment by EvanSimpson** [comment]
Generally, it seems you just need to take better care naming (and remembering) variables in the future. And always remember to validate that your application works locally before deploying and submitting!

File: classes/class3/homework/routes/cats.js [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+    if (err) {
+      console.log('Unable to find/remove cat');
+    } else {
+      console.log('Cat removed');
+      res.render('cats', {cats: cats});
+    }
+  })
+}
+
+module.exports.all = all;
```
Comment (ThomasTheDane) [inline classes/class3/homework/routes/cats.js:87]: You can store your functions in an object and then export just a single thing

**Pull request closed without merging**